### PR TITLE
Fsm update - rename jesd init functions

### PR DESCRIPTION
**Pull request by danmois**
## Pull Request Description

Remove the option to not use JESD FSM for the fmcdaq2 project and rename the jesd init functions (add legacy suffix for non-FSM functions).

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

**Review by amiclaus — APPROVED** [review]

File: projects/fmcdaq2/src/app/fmcdaq2.c [hunk 1]
```diff
@@ -79,12 +79,11 @@
 #include "xilinx_uart.h"
 #endif
 
-#ifdef JESD_FSM_ON
 #include "no_os_print_log.h"
 #include "no_os_alloc.h"
 #include "jesd204.h"
 #include "jesd204_clk.h"
-#endif
+
```
Comment (buha) [inline projects/fmcdaq2/src/app/fmcdaq2.c:86]: remove this line
i beliece this commit should be all red, no green :D

Comment (danmois) [reply projects/fmcdaq2/src/app/fmcdaq2.c]: Removed

**Review by buha — APPROVED** [review]

File: projects/fmcdaq2/src/app/fmcdaq2.c (hunk 1 above)
Comment (buha) [reply projects/fmcdaq2/src/app/fmcdaq2.c:86]: actually i take this back, i misread the diff...

**Comment by buha** [comment]
merge this as soon as CI is green

**Pull request merged**